crates/win-api-wrappers/src/netmgmt.rs

```diff
@@ -89,20 +86,45 @@ pub fn get_local_group_members(group_name: String) -> anyhow::Result<Vec<Sid>> {
         }
 
         std::slice::from_raw_parts(
-            buffer as *const u8 as *const LOCALGROUP_MEMBERS_INFO_0,
+            group_members_ptr as *const u8 as *const LOCALGROUP_MEMBERS_INFO_0,
             entries_read as usize,
         )
     };
 
+    // SAFETY: `group_members_ptr` is always a valid pointer on success.
+    let _group_members_ptr = unsafe { NetMgmtMemory::new(group_members_ptr.cast()) };
+
     let mut group_members = Vec::<Sid>::with_capacity(group_members_slice.len());
     for member in group_members_slice {
         group_members.push(Sid::try_from(member.lgrmi0_sid)?);
     }
 
-    // SAFETY: `buffer` is valid and points to memory allocated by `NetLocalGroupGetMembers`
-    unsafe {
-        NetApiBufferFree(Some(buffer.cast()));
+    Ok(group_members)
+}
+
+/// RAII wrapper for Network Management memory.
+struct NetMgmtMemory(*mut core::ffi::c_void);
+
+impl NetMgmtMemory {
+    // SAFETY: `ptr` must be a valid pointer to memory allocated by Network Management.
+    unsafe fn new(ptr: *mut core::ffi::c_void) -> Self {
+        Self(ptr)
+    }
+}
+
+impl Drop for NetMgmtMemory {
+    fn drop(&mut self) {
+        if self.0.is_null() {
+            return;
+        }
+
+        // SAFETY: No preconditions.
+        unsafe { NetApiBufferFree(Some(self.0)) };
```

Comment:
suggestion: In term of wording, I prefer
```suggestion
        // SAFETY: FFI call with no outstanding precondition.
        unsafe { NetApiBufferFree(Some(self.0)) };
```

Because there are typically preconditions, but it’s the "FFI usual business". The pointer must be valid, etc.